src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/MessageBody.cs

```diff
@@ -234,12 +286,13 @@ protected virtual void OnConsumedBytes(int count)
             var connection = headers.HeaderConnection.ToString();
```

Comment:
Wouldn't it be more efficient to `foreach` the HeaderConnection? Instead of [string.Joining](https://github.com/aspnet/Common/blob/cccd998d7a5ca0aac6e6a57869d77725409b96cc/src/Microsoft.Extensions.Primitives/StringValues.cs#L98) like we do in `ToString()`?